### PR TITLE
test: add tests for rodcutting.go

### PR DESCRIPTION
**Pull request by vil02**
#### Description of Change

I have added tests for [`rodcutting.go`](https://github.com/TheAlgorithms/Go/blob/51f0485f9289676e23d4c7cbf4a868e4704a27d6/dynamic/rodcutting.go). The test data is verified using
- https://sites.radford.edu/~nokie/classes/360/dp-rod-cutting.html
- https://www.geeksforgeeks.org/cutting-a-rod-dp-13/

Tests check both existing implementations of the solution.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/Go/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [GoDoc guidelines](https://blog.golang.org/godoc)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/Go/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
Added tests for [`rodcutting.go`](https://github.com/TheAlgorithms/Go/blob/51f0485f9289676e23d4c7cbf4a868e4704a27d6/dynamic/rodcutting.go).

### CODE REVIEW

File: dynamic/rodcutting_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	length   int
+	expected int
+}{
+	{[]int{0, 1, 5, 8, 9}, 4, 10},
+	{[]int{0, 2, 5, 7, 8, 0}, 5, 12},
+	{[]int{0, 1, 5, 8, 9, 10, 17, 17, 20}, 8, 22},
+	{[]int{0, 3, 5, 8, 9, 10, 17, 17, 20}, 8, 24},
+}
+
+func CutRobSolutionFunctionTemplate(t *testing.T, cutRodSolutionFunction func([]int, int) int) {
```
Comment (tjgurwara99) [inline dynamic/rodcutting_test.go:18]:
```suggestion
func cutRobSolTestFunc(t *testing.T, cutRodSolFunc func([]int, int) int) {
```

File: dynamic/rodcutting_test.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package dynamic
```
Comment (tjgurwara99) [inline dynamic/rodcutting_test.go:1]:
```suggestion
package dynamic_test
```

Comment (tjgurwara99) [reply dynamic/rodcutting_test.go:1]: This would result in you having to use the package `dynamic` namespace.

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: dynamic/rodcutting_test.go (hunk 1 above)
Comment (vil02) [reply dynamic/rodcutting_test.go]: Done. Sorry, I do not know how to apply your suggestion and rename the other stuff in a single commit.

File: dynamic/rodcutting_test.go [hunk 3]
```diff
@@ -1,33 +1,41 @@
...
-	{[]int{0, 1, 5, 8, 9}, 4, 10},
-	{[]int{0, 2, 5, 7, 8, 0}, 5, 12},
-	{[]int{0, 1, 5, 8, 9, 10, 17, 17, 20}, 8, 22},
-	{[]int{0, 3, 5, 8, 9, 10, 17, 17, 20}, 8, 24},
 }
 
-func CutRobSolutionFunctionTemplate(t *testing.T, cutRodSolutionFunction func([]int, int) int) {
-	for _, tc := range rodCuttingExamples {
-		actual := cutRodSolutionFunction(tc.price, tc.length)
+func getRodCuttingExamples() []RodCuttingExample {
```
Comment (tjgurwara99) [inline dynamic/rodcutting_test.go]:
```suggestion
func getTestCases() []RodCuttingExample {
```

File: dynamic/rodcutting_test.go [hunk 4]
```diff
@@ -1,33 +1,41 @@
...
+package dynamic_test
 
 import (
 	"testing"
+
+	"github.com/TheAlgorithms/Go/dynamic"
 )
 
-var rodCuttingExamples = []struct {
+type RodCuttingExample struct {
```
Comment (tjgurwara99) [inline dynamic/rodcutting_test.go]:
```suggestion
type testCase struct {
```

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

**Comment by vil02** [comment]
@tjgurwara99 thanks for the feedback.

File: dynamic/rodcutting_test.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+func getTestCases() []testCase {
+	return []testCase{
+		{[]int{0, 1, 5, 8, 9}, 4, 10},
+		{[]int{0, 2, 5, 7, 8, 0}, 5, 12},
+		{[]int{0, 1, 5, 8, 9, 10, 17, 17, 20}, 8, 22},
+		{[]int{0, 3, 5, 8, 9, 10, 17, 17, 20}, 8, 24},
+	}
+}
+
+func cutRobSolTestFunc(t *testing.T, cutRodSolFunc func([]int, int) int) {
```
Comment (tjgurwara99) [inline dynamic/rodcutting_test.go:24]:
```suggestion
func cutRodSolTestFunc(t *testing.T, cutRodSolFunc func([]int, int) int) {
```

File: dynamic/rodcutting_test.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
+package dynamic_test
+
+import (
+	"testing"
+
+	"github.com/TheAlgorithms/Go/dynamic"
+)
+
+type testCaseRodCutting struct {
```
Comment (tjgurwara99) [inline dynamic/rodcutting_test.go:9]:
```suggestion
type rodCuttingTestCase struct {
```

Comment (tjgurwara99) [reply dynamic/rodcutting_test.go:9]: For being consistent with your naming, example: `getRodCuttingTestCases`

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]
Other than that everything looks good to me.

File: dynamic/rodcutting_test.go [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+		{[]int{0, 3, 5, 8, 9, 10, 17, 17, 20}, 8, 24},
+	}
+}
+
+func cutRodSolTestFunc(t *testing.T, cutRodSolFunc func([]int, int) int) {
+	for _, tc := range getRodCuttingTestCases() {
+		actual := cutRodSolFunc(tc.price, tc.length)
+		if actual != tc.expected {
+			t.Errorf(
+				"for the price: %d and length %d expected %d, but got: %d",
```
Comment (tjgurwara99) [inline dynamic/rodcutting_test.go:29]: Non blocking suggestion. In Go, the `test` logs are simple like this

```suggestion
				"expected: %d,  got: %d",
```

**Review by tjgurwara99 — DISMISSED** [review]

**Review by tjgurwara99 — APPROVED** [review]
LGTM. Thanks!

**Comment by tjgurwara99** [comment]
For some reason I'm don't see the option to run the workflows here. cc @siriak could you check the repo settings to see that I can approve the CI runs manually?

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**